src/xtgeo/xyz/_polygons_oper.py

```diff
@@ -0,0 +1,284 @@
+        auto_alpha = np.mean(dist)
+        logger.info("Proposed auto alpha is %s", auto_alpha)
+
+        # now try this auto_alpha, and iterate to a gradually larger alpha proposal
+        # if still no success in creating a boundary
+        try:
+            return_values = _create_boundary_polygon(
+                xvec, yvec, zvec, auto_alpha * alpha_factor
+            )
+        except BoundaryError as berr:
+            # iterate and propose a new alpha value while raising an exception
+            propose_alpha = _propose_new_alpha(
+                xvec, yvec, zvec, alpha_factor, auto_alpha
+            )
+            propose_alpha_factor = alpha_factor * propose_alpha / auto_alpha
+            msg = (
+                "Your alpha_factor or alpha is too low. Either set the alpha_factor to "
+                f"approx. {propose_alpha_factor:.3f}, or set alpha to approx. "
+                f"{propose_alpha}. Setting the alpha_factor (not alpha) is "
+                "recommended\n"
+            )
+            raise BoundaryError(msg) from berr
+    else:
+        return_values = _create_boundary_polygon(xvec, yvec, zvec, alpha * alpha_factor)
+
+    # return the class parameters to populate the Polygons instance
+    if return_values is None:
+        raise BoundaryError("Cannot create a a boundary, too few points?")
+
+    return return_values
+
+
+def _propose_new_alpha(xvec, yvec, zvec, alpha_factor, alpha):
+    """The current combination of alpha_factor and alpha did not get produce a result.
+
+    Do an iteration here and propose a new alpha_value, given that the user insists
+    on keeping the current alpha_factor.
+    """
+    trials = 0
+    max_trials = 1000
+    multiplier = 1.1
+
+    proposed_alpha = alpha * multiplier
+    while trials < max_trials:
+        trials += 1
+        logger.info("Proposed alpha is %s in trial %s", proposed_alpha, trials)
+        try:
+            _create_boundary_polygon(xvec, yvec, zvec, proposed_alpha * alpha_factor)
+            print("Trial number:", trials)
+            return ceil(proposed_alpha)
+        except BoundaryError:
+            proposed_alpha *= multiplier
+
+    raise RuntimeError(f"Not able to estimate an alpha in {max_trials} iterations!")
+
+
+def _create_boundary_polygon(
+    xvec: np.ndarray,
+    yvec: np.ndarray,
+    zvec: np.ndarray,
+    alpha: float,
+):
+    xy = np.column_stack((xvec, yvec))
+    coords = np.column_stack((xy, zvec))
+
+    # return None if too few points
+    if len(xy) < MINIMUM_NUMBER_POINTS:
+        return None
+
+    edges = _alpha_shape(xy, alpha=alpha)
+    if not edges:
+        raise BoundaryError(
+            "Your alpha or alpha_factor value is too low, try increasing it!"
+        )
+
+    # sort edges and group into separate polygons
+    sorted_edges = _sort_edges_and_split_in_polygons(edges)
+
+    data = []
+    for pol, pol_edges in enumerate(sorted_edges):
+        for _, (edg_start, edg_stop) in enumerate(pol_edges):
```

Comment:
Still think we can drop this enumeration 🙂 
```suggestion
        for (edg_start, edg_stop) in pol_edges:
```